source/overloading.tex

```diff
@@ -1583,6 +1647,70 @@
 
 B b{(int*)0, (char*)0};         // OK, deduces \tcode{B<char*>}
 \end{codeblock}
+\end{example}
+
+\pnum
+\begin{example}
+\begin{codeblock}
+template <class T, class U> struct C {
+  C(T, U);                                      // \#1
+};
+template<class T, class U>
+  C(T, U) -> C<T, std::type_identity_t<U>>;     // \#2
+
+template<class V> using A = C<V *, V *>;
+template<std::Integral W> using B = A<W>;
+
+int i{};
+double d{};
+A a1(&i, &i);   // deduces \tcode{A<int>}
+A a2(i, i);     // error: cannot deduce \tcode{V *} from \tcode{i}
+A a3(&i, &d);   // error: \#1: cannot deduce \tcode{(V*, V*)} from \tcode{(int *, double *)}
+                // \#2: cannot deduce \tcode{A<V>} from \tcode{C<int *, double *>}
+B b1(&i, &i);   // deduces \tcode{B<int>}
+B b2(&d, &d);   // error: cannot deduce \tcode{B<W>} from \tcode{C<double *, double *>}
+\end{codeblock}
+Possible exposition-only implementation of the above procedure:
+\begin{codeblock}
+// The following concept ensures a specialization of A is deduced.
+template <class> class AA;
+template <class V> class AA<A<V>> { };
+template <class T> concept deduces_A = requires { sizeof(AA<T>); };
+
+// \tcode{f1} is formed from the constructor \#1 of \tcode{C}, generating the following function template
+template<T, U>
+  auto f1(T, U) -> C<T, U>;
+
+// Deducing arguments for \tcode{C<T, U>} from \tcode{C<V *, V*>} deduces \tcode{T} as \tcode{V *} and \tcode{U} as \tcode{V *}
+// \tcode{f1'} is obtained by transforming \tcode{f1} as described by the above procedure
+template<class V> requires deduces_A<C<V *, V *>>
+  auto f1_prime(V *, V*) -> C<V *, V *>;
+
+// \tcode{f2} is formed the deduction-guide \#2 of \tcode{C}
+template<class T, class U> auto f2(T, U) -> C<T, std::type_identity_t<U>>;
+
+// Deducing arguments for \tcode{C<T, std::type_identity_t<U>>} from \tcode{C<V *, V*>} deduces \tcode{T} as \tcode{V *}
+
+// \tcode{f2'} is obtained by transforming \tcode{f2} as described by the above procedure
```

Comment:
end with a "." (stand-alone comments tend to be complete sentences).

Reply:
Fixed.